### PR TITLE
Add `PathDigraph`

### PR DESCRIPTION
**Pull request by LRacine**
This is the path digraph as defined here: https://mathworld.wolfram.com/PathGraph.html

### CODE REVIEW

**Comment by wilfwilson** [comment]
@bspiers has also added the same functionality in #415, there named `PathGraph`. Which one shall we keep keep?

File: doc/examples.xml [hunk 1]
```diff
@@ -743,3 +743,38 @@ true
...
+    Filt="IsImmutableDigraph"/>, then the digraph will be immutable. 
+    If the optional first argument <A>filt</A> is not present, then <Ref
+      Filt="IsImmutableDigraph"/> is used by default.<P/>
+
+    <Example><![CDATA[
+gap> PathDigraph(2);
+<immutable symmetric digraph with 2 vertices, 2 edges>
+gap> PathDigraph(10);
+<immutable symmetric digraph with 10 vertices, 18 edges>
+gap> BookDigraph(IsMutable, 15);
```
Comment (mariatsalakou) [inline doc/examples.xml:773]: I think `BookDigraph(IsMutable, 15);` here was supposed to be `PathDigraph(IsMutable, 15);` . Everything else looks good to me!

**Comment by wilfwilson** [comment]
I recommend calling this `PathGraph`, as discussed on #444 🙂

**Comment by james-d-mitchell** [comment]
Closing in favour of #415

**Pull request closed without merging**